roles/swlib/tasks/gcsdirect.yml

```diff
@@ -13,111 +13,34 @@
-    else
-      f_name="{{ f.alt_name | default('x') }}"
-    fi
-    if ! unzip -l "{{ swlib_path }}/{{ f.name }}"
-      then
-        gcloud storage cp gs://{{ swlib_mount_src }}/${f_name} "{{ swlib_path }}/{{ f.name }}"
-    fi
-    {% endfor %}
-  register: shell_result
-  args:
-    executable: /bin/bash
-  with_items:
-    - "{{ gi_software }}"
-    - "{{ gi_interim_patches }}"
-  when:
-    - gi_install
-    - item.version == oracle_ver
-    - patch_only is not defined
-  changed_when: "'Date    Time    Name' not in shell_result.stdout"
-
-- name: gcsdirect | Copy GI patches from GCS to target instance
-  shell: |
-    set -o pipefail
-    if ! unzip -l "{{ swlib_path }}/{{ item.patchfile }}"
-      then
-        gcloud storage cp gs://{{ swlib_mount_src }}/{{ item.patchfile }} "{{ swlib_path }}/{{ item.patchfile }}"
-    fi
+    gcloud storage cp gs://{{ swlib_mount_src }}/{{ item }} "{{ swlib_path }}/{{ item }}"
   register: shell_result
   args:
     executable: /bin/bash
-  with_items: "{{ gi_patches }}"
-  when:
-    - gi_install
-    - item.release == oracle_rel
+  with_items: "{{ opatch_file_list | unique }}"
   changed_when: "'Date    Time    Name' not in shell_result.stdout"
 
-# overwrite OPatch zipfile on database host from gcs every time that may be cached currently in {{ swlib_path }}
-# reason to overwrite is that the OPatch zipfile in MOS is named the same, example: p6880880_190000_Linux-x86-64.zip
-# this change downloads OPatch from GCS bucket every time with negligible overhead as OPatch zip file is typically ~120M
-- name: gcsdirect | Copy OPatch update files from GCS to target instance
+- name: gcsdirect | Copy base software files using file_name or alt_name
   shell: |
     set -o pipefail
-    gcloud storage cp gs://{{ swlib_mount_src }}/{{ item.patchfile }} "{{ swlib_path }}/{{ item.patchfile }}"
-  register: shell_result
-  args:
-    executable: /bin/bash
-  with_items: "{{ opatch_patches }}"
-  when:
-    - item.release == oracle_ver
-    - oracle_rel != "base"
-  changed_when: "'Date    Time    Name' not in shell_result.stdout"
-
-# files with alt_name are stored as name for simplicity
-- name: gcsdirect | Copy RDBMS software from GCS to target instance
-  shell: |
-    set -o pipefail
-    {% for f in item.files %}
-    if gcloud storage ls gs://{{ swlib_mount_src }}/{{ f.name }} >/dev/null 2>&1; then
-      f_name="{{ f.name }}"
+    if gcloud storage ls gs://{{ swlib_mount_src }}/{{ item.file_name }} >/dev/null 2>&1; then
+      f_name="{{ item.file_name }}"
     else
-      f_name="{{ f.alt_name | default('x') }}"
+      f_name="{{ item.alt_name | default('x') }}"
     fi
-    if ! unzip -l "{{ swlib_path }}/{{ f.name }}"
-      then
-        gcloud storage cp gs://{{ swlib_mount_src }}/${f_name} "{{ swlib_path }}/{{ f.name }}"
+    if ! unzip -l "{{ swlib_path }}/${f_name}"; then
+      gcloud storage cp gs://{{ swlib_mount_src }}/${f_name} "{{ swlib_path }}/{{ item.file_name }}"
```

Comment:
If using the alt_name, we're actually now saving it using file_name;  is this an intentional change?

Reply:
Can we please quote the first file name too?